parquet-variant-compute/src/variant_array.rs

```
@@ -809,17 +800,11 @@ fn typed_value_to_variant(typed_value: &ArrayRef, index: usize) -> Variant<'_, '
             let date = Date32Type::to_naive_date(value);
             Variant::from(date)
         }
-        DataType::FixedSizeBinary(binary_len) => {
+        // 16-byte FixedSizeBinary is always UUID; all others illegal.
+        DataType::FixedSizeBinary(16) => {
             let array = typed_value.as_fixed_size_binary();
-            // Try to treat 16 byte FixedSizeBinary as UUID
-            let value = array.value(index);
-            if *binary_len == 16 {
-                if let Ok(uuid) = Uuid::from_slice(value) {
-                    return Variant::from(uuid);
-                }
-            }
             let value = array.value(index);
-            Variant::from(value)
+            Uuid::from_slice(value).map_or(Variant::Null, Variant::from)
```

Falling back to `Variant::Null` is not really correct, but follows much of the existing convention in this code.
If/when we make `VariantArray::value` fallible, this code should propagate the error on failure.

Alternatively, we could just `unwrap`, "knowing" that [Uuid::from_slice](https://docs.rs/uuid/latest/uuid/struct.Uuid.html#method.from_slice) can "only" fail if given a wrong slice size (which in theory should never happen because we know we have an array of 16-byte fixed binary values)?